daml-script/test/upgrades/coin-upgrade/CoinUpgrade.daml

```diff
@@ -0,0 +1,74 @@
+-- Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module CoinUpgrade where
+
+import Daml.Script
+import Coin_1_0_0.Coin qualified
+import Coin_2_0_0.Coin qualified
+
+template Aux
+  with
+    party : Party
+  where
+    signatory party
+
+    choice SoftFetch_Coin_1_0_0 : (Text, Coin_1_0_0.Coin.Coin)
+      with cid : ContractId Coin_1_0_0.Coin.Coin
+      controller party
+      do
+        coin <- softFetch cid
+        pure $
+          ( "soft fetch v1\n" <> show coin
+          , coin
+          )
+
+    choice SoftFetch_Coin_2_0_0 : (Text, Coin_2_0_0.Coin.Coin)
+      with cid : ContractId Coin_2_0_0.Coin.Coin
+      controller party
+      do
+        coin <- softFetch cid
+        pure $
+          ( "soft fetch v2\n" <> show coin
+          , coin
+          )
+
+create_v1_softFetch_v1 : Script ()
+create_v1_softFetch_v1 = do
+  alice <- allocateParty "alice"
+  cid <- alice `submit` createCmd Coin_1_0_0.Coin.Coin with
+    issuer = alice
+    owner = alice
+  _ <- alice `submit` createAndExerciseCmd (Aux alice) SoftFetch_Coin_1_0_0 with
+    cid = cid
+  pure ()
+
+create_v2_softFetch_v2 : Script ()
+create_v2_softFetch_v2 = do
+  alice <- allocateParty "alice"
+  cid <- alice `submit` createCmd Coin_2_0_0.Coin.Coin with
+    issuer = alice
+    owner = alice
+  _ <- alice `submit` createAndExerciseCmd (Aux alice) SoftFetch_Coin_2_0_0 with
+    cid = cid
+  pure ()
+
+create_v1_softFetch_v2 : Script ()
+create_v1_softFetch_v2 = do
+  alice <- allocateParty "alice"
+  cid <- alice `submit` createCmd Coin_1_0_0.Coin.Coin with
+    issuer = alice
+    owner = alice
+  _ <- alice `submit` createAndExerciseCmd (Aux alice) SoftFetch_Coin_2_0_0 with
+    cid = coerceContractId cid
+  pure ()
+
+create_v2_softFetch_v1 : Script ()
+create_v2_softFetch_v1 = do
+  alice <- allocateParty "alice"
+  cid <- alice `submit` createCmd Coin_2_0_0.Coin.Coin with
+    issuer = alice
+    owner = alice
+  _ <- alice `submitMustFail` createAndExerciseCmd (Aux alice) SoftFetch_Coin_1_0_0 with
```

Comment:
The expected `WronglyTypedContractSoft` error does get thrown but `submitMustFail` does not catch it, so the submission fails. I couldn't find any differences between the handling of the existing `WronglyTypedContract` error and the new one, does this mean I need to change something in Canton? @remyhaemmerle-da

Reply:
good catch @nickchapman-da. 
However, as I wrote in several other place, I think we should try to limit those kind of breaking change especially for a PoC. 
would not an engine crash be good enough in this case ?

Reply:
would not an engine crash be good enough in this case ?